src/shared/condition.c

```diff
@@ -177,6 +178,27 @@ static int condition_test_user(Condition *c) {
         return id == getuid() || id == geteuid();
 }
 
+static int condition_test_control_group_controller(Condition *c) {
+        int r;
+        CGroupMask system_mask, wanted_mask;
+
+        assert(c);
+        assert(c->parameter);
+        assert(c->type == CONDITION_CONTROL_GROUP_CONTROLLER);
+
+        r = cg_mask_supported(&system_mask);
+        if (r < 0) {
+                log_error_errno(r, "Failed to determine supported controllers: %m");
+                return -EBADF;
+        }
+
+        r = cg_mask_from_string(c->parameter, &wanted_mask);
+        if (r < 0)
+                return 0;
+
+        return (system_mask & wanted_mask) != 0;
```

Comment:
hmm, so with this it's possible to specify "ConditionControlGroupController=cpu memory" and it would hold if either cpu or memory is available. Is that what you had in mind? I am pretty sure it should be that both have to be around in that case. And that multiple controllers may be specified needs documentation in the man page.

Reply:
Hmm. I guess I envisaged passing multiple times in the vein of:

```
ConditionControlGroupController=foo
ConditionControlGroupController=bar
```

I didn't really consider the double case. I'll reevaluate.

Reply:
well, you used cg_mask_from_string() which parses a string that might list multiple controllers. If all you want is to parse one, then use CGROUP_CONTROLLER_TO_MASK(cgroup_controller_from_string()) instead.

i am fine with either